velox/functions/prestosql/ArrayPosition.cpp

```diff
@@ -37,24 +37,67 @@ void applyTypedFirstMatch(
   auto rawOffsets = baseArray->rawOffsets();
   auto indices = arrayDecoded.indices();
 
-  rows.applyToSelected([&](auto row) {
-    auto size = rawSizes[indices[row]];
-    auto offset = rawOffsets[indices[row]];
+  if (elementsDecoded.isIdentityMapping() && !elementsDecoded.mayHaveNulls() &&
+      searchDecoded.isConstantMapping()) {
+    if constexpr (std::is_same_v<bool, T>) {
+      auto rawElements = elementsDecoded.values<uint64_t>();
+
+      auto search = bits::isBitSet(searchDecoded.values<uint64_t>(), 0);
+
+      rows.applyToSelected([&](auto row) {
+        auto size = rawSizes[indices[row]];
+        auto offset = rawOffsets[indices[row]];
+
+        int i;
+        for (i = 0; i < size; i++) {
+          if (bits::isBitSet(rawElements, offset + i) == search) {
+            flatResult.set(row, i + 1);
+            break;
```

Comment:
nit : exit instead of break
```
      rows.applyToSelected([&](auto row) {
        auto size = rawSizes[indices[row]];
        auto offset = rawOffsets[indices[row]];

          for (auto i = 0; i < size; i++) {
              if (bits::isBitSet(rawElements, offset + i) == search) {
                  flatResult.set(row, i + 1);
                  return;
            }
        }
       // item not found
       flatResult.set(row, 0);
      });
    } 
```